PowerEditor/src/Parameters.cpp

```diff
@@ -4140,40 +4140,73 @@ bool NppParameters::feedStylerArray(TiXmlNode *node)
-	if (!pStyle)
+	auto addStyle = [&](const std::wstring& name,
+		const std::wstring& fgColor = L"",
+		const std::wstring& bgColor = L"",
+		const std::wstring& styleID = L"0",
+		const std::wstring& fromStyle = L"") -> int
 	{
-		TiXmlNode* eolColorkNode = globalStyleRoot->InsertEndChild(TiXmlElement(TEXT("WidgetStyle")));
-		eolColorkNode->ToElement()->SetAttribute(TEXT("name"), TEXT("EOL custom color"));
-		eolColorkNode->ToElement()->SetAttribute(TEXT("styleID"), TEXT("0"));
-		eolColorkNode->ToElement()->SetAttribute(TEXT("fgColor"), TEXT("DADADA"));
+		int result = 0;
+		const Style* pStyle = _widgetStyleArray.findByName(name);
+		if (pStyle == nullptr)
+		{
+			TiXmlNode* newStyle = globalStyleRoot->InsertEndChild(TiXmlElement(L"WidgetStyle"));
+			newStyle->ToElement()->SetAttribute(L"name", name);
+			newStyle->ToElement()->SetAttribute(L"styleID", styleID);
 
-		_widgetStyleArray.addStyler(0, eolColorkNode);
-	}
+			const Style* pStyleFrom = fromStyle.empty() ? nullptr : _widgetStyleArray.findByName(fromStyle);
+			if (pStyleFrom != nullptr)
+			{
+				constexpr size_t bufSize = 7;
+				if (!fgColor.empty())
+				{
+					wchar_t strColor[bufSize] = { '\0' };
+					swprintf(strColor, bufSize, L"%6X", rgbhex(pStyleFrom->_fgColor));
+					newStyle->ToElement()->SetAttribute(L"fgColor", strColor);
+				}
 
-	const Style* pStyleNpc = _widgetStyleArray.findByName(g_npcStyleName);
-	if (!pStyleNpc)
-	{
-		TiXmlNode* npcColorkNode = globalStyleRoot->InsertEndChild(TiXmlElement(TEXT("WidgetStyle")));
-		npcColorkNode->ToElement()->SetAttribute(TEXT("name"), g_npcStyleName);
-		npcColorkNode->ToElement()->SetAttribute(TEXT("styleID"), TEXT("0"));
+				if (!bgColor.empty())
+				{
+					wchar_t strColor[bufSize] = { '\0' };
+					swprintf(strColor, bufSize, L"%6X", rgbhex(pStyleFrom->_bgColor));
+					newStyle->ToElement()->SetAttribute(L"bgColor", strColor);
+				}
 
-		// use color from style White space symbol
-		const Style* pStyleWS = _widgetStyleArray.findByName(TEXT("White space symbol"));
-		if (pStyleWS)
-		{
-			constexpr size_t bufSize = 7;
-			wchar_t strColor[bufSize] = { '\0' };
-			swprintf(strColor, bufSize, L"%6X", rgbhex(pStyleWS->_fgColor));
-			npcColorkNode->ToElement()->SetAttribute(L"fgColor", strColor);
-		}
-		else
-		{
-			npcColorkNode->ToElement()->SetAttribute(L"fgColor", L"DADADA");
+				result = 2;
+			}
+			else
+			{
+				if (!fgColor.empty())
+				{
+					newStyle->ToElement()->SetAttribute(L"fgColor", fgColor);
+				}
+
+				if (!bgColor.empty())
+				{
+					newStyle->ToElement()->SetAttribute(L"bgColor", bgColor);
+				}
+
+				result = 1;
+			}
+			
+
+			if (!fgColor.empty() || !bgColor.empty())
+			{
+				_widgetStyleArray.addStyler(0, newStyle);
+				return result;
+			}
+			return -1;
 		}
+		return result;
+	};
 
-		_widgetStyleArray.addStyler(0, npcColorkNode);
-	}
+	addStyle(L"Change History modified", L"FF8000", L"FF8000");

```

Comment:
missing styles will be added at end, they won't be sorted same as in fresh new installs,
but still better than searching for non-existing entry in style configurator